src/Modules/OrchardCore.Commerce/Views/Payment/Index.cshtml

```diff
@@ -0,0 +1,100 @@
+@using Money
+@using OrchardCore.Commerce.Abstractions
+@using OrchardCore.Commerce.Controllers
+@using OrchardCore.Commerce.Extensions
+@using OrchardCore.Commerce.Models
+@using OrchardCore.Commerce.ViewModels
+@using OrchardCore.Entities
+@using OrchardCore.Settings
+
+@inject IShoppingCartPersistence ShoppingCartPersistence
+@inject IPriceService PriceService
+@inject IPriceSelectionStrategy PriceSelectionStrategy
+@inject ISiteService SiteService
+
+@{
+    // Shopping cart ID is null by default currently.
+    var currentShoppingCart = await ShoppingCartPersistence.RetrieveAsync();
+
+    var cartIsNotEmpty = currentShoppingCart.Count != 0;
+
+    IEnumerable<Amount> totals;
+    Amount defaultTotal = new();
+    string publishableKey = string.Empty;
+
+    if (cartIsNotEmpty)
+    {
+        totals = await currentShoppingCart.CalculateTotalsAsync(PriceService, PriceSelectionStrategy);
+
+        // We can only pay in one currency at a time. Later we have to figure out what to do if there are multiple
+        // totals i.e., multiple currencies.
+        defaultTotal = totals.FirstOrDefault();
```

Comment:
Doesn't this mean that right now if you have multiple currencies, you buy the whole cart but only pay for the line items in the first currency? Might be safer to use `SingleOrDefault` (or `Single`) for the time being. Then if someone needs it before a solution is found, they can file a bug. 😉

Reply:
> Doesn't this mean that right now if you have multiple currencies, you buy the whole cart but only pay for the line items in the first currency?

Yes, that's correct. This needs to be fixed, but I think later, since this needs to be discussed too first.